matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/crypto/OlmMachine.kt

```diff
@@ -647,6 +660,25 @@ internal class OlmMachine(
         return userMap
     }
 
+    /**
+     * If the user is untracked or forceDownload is set to true, a key query request will be made.
+     * It will suspend until query response, and the device list will be returned.
+     * As there is no API to know tracking status, we consider that if there are no known devices the user is un tracked.
+     *
+     * The key query request will be retried a few time in case of shaky connection, but could fail.
+     */
+    suspend fun ensureUserDevicesMap(userIds: List<String>, forceDownload: Boolean = false): MXUsersDevicesMap<CryptoDeviceInfo> {
+        val known = getUserDevicesMap(userIds)
+        return if (known.isEmpty /* We have no api to know if was tracked..*/ || forceDownload) {
```

Comment:
I think we'll add a method to check if a user is tracked, but that can come later as an optimization.